### PR TITLE
Add an e2e test for the source

### PR DESCRIPTION
**Pull request by mattmoor**
Based on: https://github.com/vmware-tanzu/sources-for-knative/pull/34

### CODE REVIEW

File: config/200-clusterrole.yaml [hunk 1]
```diff
@@ -44,3 +44,20 @@ rules:
...
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vmw-builtin-podspecable-binding
+  labels:
+    sources.tanzu.vmware.com/release: devel
+    duck.knative.dev/podspecable: "true"
+rules:
+  - apiGroups: ["apps"]
+    resources: ["deployments", "daemonsets", "statefulsets", "replicasets"]
```
Comment (vaikas) [inline config/200-clusterrole.yaml]: this is for the bindings stuff perhaps we should add a comment for that?

File: test/clients.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import (
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"knative.dev/pkg/test"
+
+	// Support running e2e on GKE
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```
Comment (vaikas) [inline test/clients.go]: do you want to always import this?

File: test/e2e/util.go [hunk 3]
```diff
@@ -150,3 +152,196 @@ func RunJobScript(t *testing.T, clients *test.Clients, image, script string, sel
...
+	t.Log("", "job", spew.Sprint(job))
+
+	cancel := func() {
+		err := clients.KubeClient.Kube.BatchV1().Jobs(job.Namespace).Delete(job.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error cleaning up Job %s", job.Name)
+		}
+	}
+	waiter := func() {
+		// Wait for the Pod to show up in the Endpoints resource.
```
Comment (vaikas) [inline test/e2e/util.go:215]: this comment is wrong? Left over cut&paste?

File: test/clients.go (hunk 2 above)
Comment (mattmoor) [reply test/clients.go]: It's test code, so I don't think it really affects/hurts application side, so yolo.

File: test/e2e/util.go (hunk 3 above)
Comment (mattmoor) [reply test/e2e/util.go:215]: good catch.

**Review by vaikas — APPROVED** [review]

**Pull request merged**